packages/elements/src/globals/internal/link.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { classMap } from 'lit-html/directives/class-map';
+import { html, property, query, LitElement } from 'lit-element';
+import settings from 'carbon-components/es/globals/js/settings';
+import ifNonNull from 'carbon-custom-elements/es/globals/directives/if-non-null';
+import FocusMixin from 'carbon-custom-elements/es/globals/mixins/focus';
+import styles from 'carbon-custom-elements/scss/components/link/link.scss';
+
+const { prefix } = settings;
+
+/**
+ * Link.
+ */
+class DDSLinkWithIcon extends FocusMixin(LitElement) {
```

Comment:
hey Akira, i might need a walkthrough on this one, this is an internal `link` file but seems to be specifically about the LinkWithIcon.